### ISSUE TITLE
How to access BOSH Director on GCP when concourse-up deploy fails deploying Concourse?

### ISSUE THREAD

**Issue by gerhard** [EngineerBetter/concourse-up#83]
`concourse-up deploy ci --iaas GCP --region europe-west1` fails with:

```
Task 10 | 16:48:55 | Preparing deployment: Preparing deployment (00:00:01)
Task 10 | 16:48:56 | Preparing deployment: Rendering templates (00:00:01)
Task 10 | 16:48:58 | Preparing package compilation: Finding packages to compile (00:00:01)
Task 10 | 16:48:59 | Creating missing vms: web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0)
Task 10 | 16:48:59 | Creating missing vms: worker/cf010b93-25f5-4c61-b7f2-c865a71c36ac (0)
Task 10 | 16:48:59 | Creating missing vms: worker/8f7677d0-a97a-44b3-bb12-82ddf127680d (1)
Task 10 | 16:49:52 | Creating missing vms: web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0) (00:00:53)
Task 10 | 16:50:16 | Creating missing vms: worker/cf010b93-25f5-4c61-b7f2-c865a71c36ac (0) (00:01:17)
Task 10 | 16:50:16 | Creating missing vms: worker/8f7677d0-a97a-44b3-bb12-82ddf127680d (1) (00:01:17)
Task 10 | 16:50:17 | Updating instance web: web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0) (canary)
Task 10 | 16:50:17 | Updating instance worker: worker/cf010b93-25f5-4c61-b7f2-c865a71c36ac (0) (canary) (00:01:02)
Task 10 | 16:51:19 | Updating instance worker: worker/8f7677d0-a97a-44b3-bb12-82ddf127680d (1) (00:00:43)
Task 10 | 17:02:13 | Updating instance web: web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0) (canary) (00:11:56)
                   L Error: 'web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0)' is not running after update. Review logs for failed jobs: atc, grafana
Task 10 | 17:02:13 | Error: 'web/27568ac3-125a-4ace-8d19-a8e37b60e55c (0)' is not running after update. Review logs for failed jobs: atc, grafana
```

How can I access the BOSH Director to see what the failure is?

**Comment by crsimmons** [comment]
@gerhard You should be able to export the necessary env vars using `eval "$(concourse-up info ci --iaas GCP --region europe-west1 --env)"

Failing that the director credentials should be in one of the files in your `concourse-up-ci-europe-west1-config` GCS bucket.

**Comment by gerhard** [comment]
That `concourse-up info` is failing on:

https://github.com/EngineerBetter/concourse-up/blob/41dff501f9dccd77580462122520b73f79bb5d86/concourse/info.go#L176-L179

`config.json` in GCS booked doesn't look complete, `director_{ca_cert,cert,key}` values are empty:

```json
{
  "allow_ips": "\"0.0.0.0/0\"",
  "availability_zone": "europe-west1-b",
  "concourse_ca_cert": "",
  "concourse_cert": "",
  "concourse_db_name": "concourse_atc",
  "concourse_key": "",
  "concourse_password": "",
  "concourse_username": "",
  "concourse_user_provided_cert": false,
  "concourse_web_size": "small",
  "concourse_worker_count": 2,
  "concourse_worker_size": "xlarge",
  "config_bucket": "concourse-up-37x-europe-west1-config",
  "credhub_admin_client_secret": "",
  "credhub_ca_cert": "",
  "credhub_password": "",
  "credhub_url": "",
  "credhub_username": "",
  "deployment": "concourse-up-37x",
  "director_ca_cert": "",
  "director_cert": "",
  "director_hm_user_password": "***",
  "director_key": "",
  "director_mbus_password": "***",
  "director_nats_password": "***",
  "director_password": "***",
  "director_public_ip": "",
  "director_registry_password": "***",
  "director_username": "admin",
  "domain": "37x.concourse.rabbitmq.com",
  "encryption_key": "***",
  "github_auth_is_set": true,
  "github_client_id": "***",
  "github_client_secret": "***",
  "grafana_password": "",
  "grafana_username": "",
  "hosted_zone_id": "concourse",
  "hosted_zone_record_prefix": "37x",
  "iaas": "GCP",
  "multi_az_rds": false,
  "namespace": "europe-west1",
  "private_key": "***",
  "project": "37x",
  "public_key": "***",
  "rds_default_database_name": "bosh-ccesxzzt",
  "rds_instance_class": "db-g1-small",
  "rds_password": "***",
  "rds_username": "***",
  "region": "europe-west1",
  "source_access_ip": "82.39.214.211",
  "spot": true,
  "tags": null,
  "tf_state_path": "terraform.tfstate",
  "version": "",
  "worker_type": "m4"
}
```

**Comment by crsimmons** [comment]
There should be files called `director-creds.yml` and `director-state.json` in your bucket that are updated after `create-env`.

It looks like config was not being saved after the first successful `create-env` which is why yours was mostly empty. We've made a change to save it to the bucket even if the initial Concourse deployment fails. You can try building using https://github.com/EngineerBetter/concourse-up/commit/e56b71a8472b914c2f050ceb380029aa8a16b485

**Comment by gerhard** [comment]
Trying now, thank you!

**Comment by gerhard** [comment]
Now that I've managed to `bosh ssh` into the `concourse/web` instance, I can see why `atc` & `grafana` jobs are failing:

```
tail /var/vcap/sys/log/{atc,grafana}/*.log
==> /var/vcap/sys/log/atc/atc.stderr.log <==
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input
tls: failed to find any PEM data in certificate input

==> /var/vcap/sys/log/atc/atc.stdout.log <==

==> /var/vcap/sys/log/grafana/grafana.log <==
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing HTTP Server" logger=http.server address=0.0.0.0:3000 protocol=https subUrl= socket=
t=2019-01-22T14:10:57+0000 lvl=eror msg="Fail to start server" logger=server error="tls: failed to find any PEM data in certificate input"
t=2019-01-22T14:10:57+0000 lvl=info msg="Shutdown started" logger=server code=1 reason="Startup failed"
t=2019-01-22T14:10:57+0000 lvl=info msg="stopped http server" logger=http.server
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing Alerting" logger=alerting.engine
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing CleanUpService" logger=cleanup
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped CleanUpService" logger=cleanup reason="context canceled"
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped Stream Manager"
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped Alerting" logger=alerting.engine reason="context canceled"
t=2019-01-22T14:10:57+0000 lvl=info msg="Shutdown completed" logger=server reason="context canceled"

==> /var/vcap/sys/log/grafana/stderr.log <==

==> /var/vcap/sys/log/grafana/stdout.log <==
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing HTTP Server" logger=http.server address=0.0.0.0:3000 protocol=https subUrl= socket=
t=2019-01-22T14:10:57+0000 lvl=eror msg="Fail to start server" logger=server error="tls: failed to find any PEM data in certificate input"
t=2019-01-22T14:10:57+0000 lvl=info msg="Shutdown started" logger=server code=1 reason="Startup failed"
t=2019-01-22T14:10:57+0000 lvl=info msg="stopped http server" logger=http.server
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing Alerting" logger=alerting.engine
t=2019-01-22T14:10:57+0000 lvl=info msg="Initializing CleanUpService" logger=cleanup
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped CleanUpService" logger=cleanup reason="context canceled"
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped Stream Manager"
t=2019-01-22T14:10:57+0000 lvl=info msg="Stopped Alerting" logger=alerting.engine reason="context canceled"
t=2019-01-22T14:10:57+0000 lvl=info msg="Shutdown completed" logger=server reason="context canceled"
```

Is this a known error? Digging further...

**Comment by crsimmons** [comment]
This has been raised before on the AWS side (https://github.com/EngineerBetter/concourse-up/issues/77). We haven't gotten around to looking at it so PRs would be welcome!

**Comment by gerhard** [comment]
Closing this in favour of #77 . Using values instead of paths made the Concourse deployment on GCP go through. Thank you!

**Comment by gerhard** [comment]
Any chance you could cut a new release with all the latest GCP-related fixes, maybe `0.17.1`?

I am currently using a dev build, would be nice to depend on an official release.

**Comment by gerhard** [comment]
That's super duper, switching to it now. 💯
